### ISSUE TITLE
Feature Request: Add Metatensor wrapper

### ISSUE THREAD

**Issue by PicoCentauri** [Luthaf/vesin#20]
The [metatensor library](https://docs.metatensor.org/) is becoming a widely used library in the field of atomistic machine learning. Its atomistic part has support for storing neighbor list data alongside with structure data. Similar to the ASE wrapper it may be useful to also have a function that (a) returns a neighbor list in the metatensor format and/or (b) directly attach a neighbor list to a metatensor `system`.

I am thinking about an API from the Python side for (a):

```python 
import vesin 
from metatensor.torch.atomistic import System,

system = System(...)

neighbors = vesin.metatensor_neighbor_list(system, cutoff=3.0, full_list=False)
```

where `neighbors` is a `TensorBlock` in the correct format to be used to be attached with `add_neighbor_list`

and as I suggested in (b) one may also directly attach it to a system

```python 
import vesin 
from metatensor.torch.atomistic import NeighborListOptions , System,

system = System(...)
nl_options = NeighborListOptions(cutoff=3.0, full_list=False)

vesin.add_metatensor_neighbor_list(system, nl_options)
```

**Comment by ceriottm** [comment]
I'm looking into this as a way to reduce the boilerplate code that is infesting `torch-pme` as much as more of our projects.
I'm thinking that the cleanest thing to do would be to build a separate `vesin[metatensor]`, that provides a `NeighborList` object that takes a `NeighborListOptions` upon creation, and then `nl.compute(system)` will compute the nl and add it to `system`. Does everyone like this and /or sees no major issues with the design?

**Comment by Luthaf** [comment]
that looks good to me, it could even just be in `vesin.torch.metatensor`; and does not have to be as a separate python package or even use separate installation instructions

**Comment by Luthaf** [comment]
Fixed by #23
